geth/account/accounts_test.go

```diff
@@ -109,3 +112,154 @@ func TestVerifyAccountPasswordWithAccountBeforeEIP55(t *testing.T) {
 }
+
+// newTestManager returns the account manager and the mock node manager
+// that's being used within it for testing purposes.
+func newTestManager(t *testing.T) (accManager *account.Manager, nodeManager *common.MockNodeManager) {
+	ctrl := gomock.NewController(t)
+	nodeManager = common.NewMockNodeManager(ctrl)
+	accManager = account.NewManager(nodeManager)
+	return
+}
+
+// newTestKeyStore returns a key store to use in tests
+func newTestKeyStore(t *testing.T, dir string) (keyStore *keystore.KeyStore, keyStoreDir string) {
+	var err error
+	keyStoreDir, err = ioutil.TempDir(os.TempDir(), dir)
+	require.NoError(t, err)
+	keyStore = keystore.NewKeyStore(keyStoreDir, keystore.LightScryptN, keystore.LightScryptP)
+	return
+}
+
+func TestCreateAndRecoverAccountSuccess(t *testing.T) {
+	accManager, nodeManager := newTestManager(t)
+
+	var password string
+
+	keyStore, keyStoreDir := newTestKeyStore(t, "accounts")
+	defer os.RemoveAll(keyStoreDir) //nolint: errcheck
+
+	// Don't fail on empty password
+	nodeManager.EXPECT().AccountKeyStore().Return(keyStore, nil)
+	_, _, _, err := accManager.CreateAccount(password)
+	require.NoError(t, err)
+
+	password = "some-pass"
+
+	nodeManager.EXPECT().AccountKeyStore().Return(keyStore, nil)
+	addr1, pubKey1, mnemonic, err := accManager.CreateAccount(password)
+	require.NoError(t, err)
+
+	require.NotNil(t, addr1)
+	require.NotNil(t, pubKey1)
+	require.NotNil(t, mnemonic)
+
+	// Now recover the account using the mnemonic seed and the password
+	nodeManager.EXPECT().AccountKeyStore().Return(keyStore, nil)
+	addr2, pubKey2, err := accManager.RecoverAccount(password, mnemonic)
+	require.NoError(t, err)
+
+	require.Equal(t, addr1, addr2)
+	require.Equal(t, pubKey1, pubKey2)
+}
+
+func TestCreateAndRecoverAccountFail_KeyStore(t *testing.T) {
+	accManager, nodeManager := newTestManager(t)
+
+	password := "some-pass"
+
+	keyStore, keyStoreDir := newTestKeyStore(t, "accounts")
+	defer os.RemoveAll(keyStoreDir) //nolint: errcheck
+
+	expectedErr := errors.New("Non-nil error string")
+	nodeManager.EXPECT().AccountKeyStore().Return(nil, expectedErr)
+	_, _, _, err := accManager.CreateAccount(password)
+	require.Equal(t, err, expectedErr)
+
+	// Create a new account to use the mnemonic seed.
+	nodeManager.EXPECT().AccountKeyStore().Return(keyStore, nil)
+	_, _, mnemonic, err := accManager.CreateAccount(password)
+	require.NoError(t, err)
+
+	nodeManager.EXPECT().AccountKeyStore().Return(nil, expectedErr)
+	_, _, err = accManager.RecoverAccount(password, mnemonic)
+	require.Equal(t, err, expectedErr)
+}
+
+func TestSelectAccount(t *testing.T) {
+	accManager, nodeManager := newTestManager(t)
+
+	password := "some-pass"
+
+	keyStore, keyStoreDir := newTestKeyStore(t, "accounts")
+	defer os.RemoveAll(keyStoreDir) //nolint: errcheck
+
+	nodeManager.EXPECT().AccountKeyStore().Return(keyStore, nil)
+	addr, _, _, err := accManager.CreateAccount(password)
+	require.NoError(t, err)
+
+	w := whisper.New(nil)
```

Comment:
In this context, `w` may be hard to figure out what it refers to. I would recommend calling it `shh` which is usually used for Whisper service.